### PR TITLE
refactor(git): extract _get_commit_command

### PR DESCRIPTION
**Pull request by bearomorphism**
## Checklist

- [ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [x] Add test cases to all the changes you introduce
- [x] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [x] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly

### CODE REVIEW

**Comment by bearomorphism** [comment]
I think I encountered the same issue:

https://github.com/commitizen-tools/commitizen/pull/740#issuecomment-1551425158

File: commitizen/git.py [hunk 1]
```diff
@@ -119,19 +119,22 @@ def commit(
...
-    elif committer_date:
-        command = f"GIT_COMMITTER_DATE={committer_date} {command}"
-
+    command = _get_commit_command(args, committer_date, f.name)
     c = cmd.run(command)
     os.unlink(f.name)
     return c
 
 
+def _get_commit_command(args: str, committer_date: str | None, name: str) -> str:
```
Comment (Lee-W) [inline commitizen/git.py:128]:
```suggestion
def _generate_commit_command(args: str, committer_date: str | None, name: str) -> str:
```

it might be more accurate this way. We're not getting it from anywhere. (yep, but it sometimes make me feel like it's a generator...)

File: commitizen/git.py [hunk 2]
```diff
@@ -119,19 +119,22 @@ def commit(
...
+    command = f'git commit {args} -F "{name}"'
+    if not committer_date:
+        return command
+    if os.name != "nt":
+        return f"GIT_COMMITTER_DATE={committer_date} {command}"
+    # Using `cmd /v /c "{command}"` sets environment variables only for that command
+    return f'cmd /v /c "set GIT_COMMITTER_DATE={committer_date}&& {command}"'
```
Comment (Lee-W) [inline commitizen/git.py:135]: We probably could mock `os` and add test cases to improve this coverage

File: commitizen/git.py (hunk 1 above)
Comment (Lee-W) [reply commitizen/git.py]: sounds like a good option

**Review by Lee-W — APPROVED** [review]
